File: 백준/Bronze/2775. 부녀회장이 될테야/부녀회장이 될테야.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.stream.IntStream;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));


        int T = Integer.parseInt(br.readLine());


        while (T --> 0) {
            int k = Integer.parseInt(br.readLine());  // 층
            int n = Integer.parseInt(br.readLine());  // 호

            int[] apart = new int[n];

            IntStream.range(1, n + 1).forEach(i -> apart[i - 1] = i);
//            Arrays.stream(apart).forEach(System.out::println);

            for (int x = 0; x < k; x++){
                for (int y = 1; y < n; y++) {
                    apart[y] += apart[y - 1];
                }
            }
            System.out.println(apart[n - 1]);

        }
    }
}

File: 백준/Bronze/2775. 부녀회장이 될테야/부녀회장이 될테야.py
T = int(input())

for _ in range(T):
    k = int(input())  # 층
    n = int(input())  # 호
    apart = [[0] * n for _ in range(k + 1)]
    for i in range(k + 1):
        apart[i][0] = 1

    for j in range(n):
        apart[0][j] = j + 1

    for x in range(1, k + 1):
        for y in range(1, n):
            apart[x][y] = sum(apart[x - 1][:y + 1])
    print(apart[k][n-1])





